### PR TITLE
Clipboard paste

### PR DESCRIPTION
**Pull request by Reettaphant**
## What's changed?

_Detail the main feature of this PR and optionally a link to the relevant issue / card_

- Fixes a bug where clipboard paste was not working because we were passing the wrong argument to it after drag/drop refactor. 

- Changes the clipboard paste behaviour, we now can past onto the clipboard with command +v without focusing on the clipboard first. I've checked and this does affect behaviour to paste text to input fields.

## Implementation notes

_Include any specific areas you want to highlight for review that you feel might be worthy of discussion (i.e. any non-obvious decisions you've made)_

@jonathonherbert Don't know if you have an opinion on a better/neater place where the new paste function could go. 

### General
- [ ] 🤖 Relevant tests added
- [x] ✅ CI checks / tests run locally
- [ ] 🔍 Checked on CODE

### Client
- [x] 🚫 No obvious console errors on the client (i.e. React dev mode errors)
- [x] 🎛️ No regressions with existing user interactions (i.e. all existing buttons, inputs etc. work)
- [ ] 📷 Screenshots / GIFs of relevant UI changes included

### CODE REVIEW

**Comment by ajwl** [comment]
In order to user test this - what are the steps to replicate this journey? 
(I tried Cmd+C on a story and Cmd+V on Clipboard but this appeared not to work for me. Do I need to hit some keyboard setting or something?)

**Comment by paperboyo** [comment]
This may only pertain to d&d copy/paste, but @ajwl brings attention to a lack of (useful) parity of v2: in v1 a simple cmd+v pastes a link to the clipboard regardless of current app focsu. This, sadly, doesn’t work in v2.

**Comment by jonathonherbert** [comment]
@ajwl I think the behaviour relates to copying a link from elsewhere -- e.g. copy article link in browser, ctrl-v whilst focused on fronts tool, article appears in clipboard. Not to say the behaviour you are describing wouldn't be nice -- (although we do have a + arrow to add articles to clipboard as a workaround)

@paperboyo I think the behaviour you're describing is what's implemented in this PR -- `Changes the clipboard paste behaviour, we now can past onto the clipboard with command +v without focusing on the clipboard first`

**Comment by paperboyo** [comment]
Bombastic! Always happy to learn I’m an idiot!

**Comment by ajwl** [comment]
@jonathonherbert ah thanks for explanation. That does work with external links.

**Comment by jonathonherbert** [comment]
@Reettaphant this looks good to me -- one thing to mention, the focus state for the clipboard component itself is now redundant as we don't require it to be focused for a paste operation. Would it be worth removing it so we don't mislead users by suggesting that it does something useful?

**Comment by ajwl** [comment]
👍 works!

File: client-v2/src/keyboard/index.ts [hunk 1]
```diff
@@ -63,7 +66,31 @@ export const createKeyboardActionMap = (store: Store): KeyboardBindingMap => ({
   'command+v': {
     title: 'Paste',
     description: 'Paste an entity',
-    action: bindActionMap(store, paste)
+    action: () =>
+      (store.dispatch as Dispatch)(async (dispatch: Dispatch) => {
```
Comment (jonathonherbert) [inline client-v2/src/keyboard/index.ts]: Me and Anna have just gone through this, and the double parenthesis are a little confusing -- we came up with something along the lines of 
```
action: async () => {
      const dispatch: Dispatch = store.dispatch;
      try {
        if (!navigator || !(navigator as any).clipboard) {
          throw new Error('No navigator available on paste');
        }
        // A temporary any here pending proper typings
        const content: string = await (navigator as any).clipboard.readText();
        if (!content) {
          return;
        }
        const contentData: RefDrop = { type: 'REF', data: content };
        const articleFragment = await dispatch(
          createArticleFragment(contentData)
        );
        if (!articleFragment) {
          return;
        }
        dispatch(
          insertClipboardArticleFragment('clipboard', 0, articleFragment.uuid)
        );
      } catch (e) {
        Raven.captureMessage(`Paste to clipboard failed: ${e.message}`);
      }
```
-- what do you think?

Comment (Reettaphant) [reply client-v2/src/keyboard/index.ts]: Yes, this is much better! I will change it.

**Comment by Reettaphant** [comment]
@jonathonherbert We still need the clipboard focus state for moving articles up and down with arrow keys so I think it makes sense to keep it there.

**Pull request merged**